fix: grammatical errors and rephrase README.md (#2939)
AleoHQ/snarkOS: README.md

    - Provers: CUDA-enabled GPU (optional)
    - Validators: Not required at this time

> Just so you know, to run an Aleo Prover that is **competitive**, the machine will need more than these requirements.

Reviewer: ```suggestion
Please note that in order to run an Aleo Prover that is **competitive**, the machine will need more than these requirements.
```
